src/modules/global.ts

```diff
@@ -65,8 +71,38 @@ const initialState: GlobalStateType = {
 
   ...DEFAULT_FACE_STATE,
   ...DEFAULT_DICE_OPTIONS,
+  ...DEFAULT_PROJECT_SETTINGS,
 }
 const globalState = createStore(reducer, initialState)
-export const useGlobalState = globalState.useGlobalState
+
+const eventManager = new EventEmitter()
+
+const GlobalStateChangeEventName = 'GlobalStateChange'
+
+export const subscribeToAllChanges = (cb: () => void): void => {
+  eventManager.addListener(GlobalStateChangeEventName, cb)
+  console.log(`Subscribed to ${GlobalStateChangeEventName}`)
+}
+
+export const unsubscribeFromAllChanges = (cb: () => void): void => {
+  eventManager.removeEvent(GlobalStateChangeEventName, cb)
+}
+
+export const useGlobalState = <S>(key: GlobalStateKey): [any, Dispatch<SetStateAction<S>>] => {
```

Comment:
I cannot figure out what type should be the "value" in the return from useGlobalState() ... I'm sure it's a feature of TypeScript that I don't understand.  I'm looking forward to learning what it is!